packages/safe-tools-client/app/components/setup-safe-modal/index.gts

```diff
@@ -12,22 +17,32 @@ interface Signature {
   Element: HTMLElement;
   Args: { 
     isOpen: boolean 
-    loading?: boolean;
     onClose: () => void;
   };
 }
 
-//TODO: replace with correct flags and logic
-const gasCost = `0.001899365 ETH (USD$3.01)`;
-const hasEnoughGas = true;
-
 export default class SetupSafeModal extends Component<Signature> {
+  //TODO: replace with correct flags and logic
+  gasCost = `0.001899365 ETH (USD$3.01)`;
+  notEnoughBalance = false;
+
+  @tracked provisioning = false;
+
+  @action async createSafe(): Promise<void> { 
+     //TODO: use sdk method createSafeWithModuleAndGuard
+    this.provisioning = true;
+    setTimeout(() => {
+      console.log('Create safe')
+      this.provisioning = false;
+      this.args.onClose();
+    }, 2000); 
+  }
+
   <template>
-    <BoxelModal
+    <BoxelModal 
       @size='medium'
-      @isOpen={{@isOpen}}
-      @onClose={{@onClose}}
-    >
+      @isOpen={{@isOpen}} 
+      @onClose={{if this.provisioning (noop) @onClose}}>
```

Comment:
The cursor on the overlay suggests that clicking it will dismiss the modal when provisioning is in progress:

![Cardstack Safe Tools 2022-10-18 17-22-53](https://user-images.githubusercontent.com/43280/196556353-9bfffc99-4478-40c1-a391-8bfbc88d45fb.png)

This makes me think maybe we need a first-class option for `Boxel::Modal` that makes it not dismissible vs this workaround-ish `noop` 🤔

Reply:
Agreed, will add a ticket for it